packages/client/src/contexts/TestContext/index.tsx

```diff
@@ -0,0 +1,67 @@
+import React, {
+  createContext,
+  ReactNode,
+  useContext,
+  useEffect,
+  useState,
+} from "react";
+import { LoadingSpinner, ServerError } from "../../components";
+
+export type TestContextType = {
+  connectionMessage?: string;
+};
+
+export type TestProviderProps = {
+  children?: ReactNode;
+};
+
+export const TestContext = createContext<TestContextType>({});
+
+export function TestProvider({ children }: TestProviderProps) {
```

Comment:
This is just a pretty bare demonstration of how [React context](https://beta.reactjs.org/learn/passing-data-deeply-with-context) works. The benefit here is that we can keep server-dependent state very high in the application, while ensuring its hydration (selectively notifying only relevant pieces of component of changed values) of child components is done any necessary render.